march_hardware/include/march_hardware/march_robot.h

```diff
@@ -43,23 +43,23 @@ class MarchRobot
   MarchRobot(MarchRobot&&) = delete;
   MarchRobot& operator=(MarchRobot&&) = delete;
 
-  void resetIMotionCubes();
+  void resetMotorControllers();
 
-  void startEtherCAT(bool reset_imc);
+  void startCommunication(bool reset_motor_controllers);
 
-  void stopEtherCAT();
+  void stopCommunication();
 
   int getMaxSlaveIndex();
 
   bool hasValidSlaves();
 
-  bool isEthercatOperational();
+  bool isCommunicationOperational();
 
-  std::exception_ptr getLastEthercatException() const noexcept;
+  std::exception_ptr getLastCommunicationException() const noexcept;
 
-  void waitForPdo();
+  void waitForUpdate();
 
-  int getEthercatCycleTime() const;
+  int getCycleTime() const;
```

Comment:
I was already hesitant about the removal of ethercat from these namings, because that indeed introduces a lot more vagueness. Hence I reverted them for now. 

I do not really understand what you say about reset and start functions; the reset of the MarchRobot directly calls the reset of the MotorController, which should work for any controller. Similarly, the start method calls start for the ethercatMaster, which in turn simply calls initialize for each MotorController. This should work for any Ethercat motor controller. For a none-ethercat motor controller, a new communication master object should be created, with another implementation of the start method.